nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -55,9 +55,24 @@ int Pooling2DLayer::initialize(Manager &manager) {
     out_dim.width(in_dim.channel());
   }
 
+  if (pooling_type == PoolingType::max) {
+    max_idx.reserve(in_dim.batch() * output_dim[0].getFeatureLen());
```

Comment:
unless you really want this to be here, please don't allocate memory in setBatch. Rather move it to the first forward/backward.

Reply:
redirecting duplicated discussion to https://github.com/nnstreamer/nntrainer/pull/1051#discussion_r603766020